test/tests/pbs_accumulate_resc_used.py

```diff
@@ -0,0 +1,1456 @@
+            self.assertTrue(s)
+            acctlog_match = 'resources_used.stra=\"glad\,elated\"\,\"happy\"'
+            s = self.server.accounting_match(
+                "E;%s;.*%s.*" % (jid, acctlog_match), regexp=True, n=100)
+            self.assertTrue(s)
+
+    def test_resource_bool(self):
+        """
+        To test that boolean value are not getting aggregated
+        """
+
+        # Create a boolean type resource
+        attr = {}
+        attr['type'] = 'boolean'
+        self.server.manager(
+            MGR_CMD_CREATE, RSC, attr,
+            id='foo_bool', runas=ROOT_USER,
+            logerr=False)
+
+        hook_body = """
+import pbs
+e=pbs.event()
+j=e.job
+if j.in_ms_mom():
+    j.resources_used["foo_bool"] = True
+else:
+    j.resources_used["foo_bool"] = False
+"""
+
+        hook_name = "epi_bool"
+        a = {'event': "execjob_epilogue", 'enabled': "True"}
+        self.server.create_import_hook(
+            hook_name,
+            a,
+            hook_body,
+            overwrite=True)
+
+        a = {'Resource_List.select': '3:ncpus=1',
+             'Resource_List.walltime': 10,
+             'Resource_List.place': 'scatter'}
+        j = Job(TEST_USER)
+        j.set_attributes(a)
+        j.set_sleep_time("5")
+        jid = self.server.submit(j)
+
+        # foo_bool is True
+        a = {'resources_used.foo_bool': "True",
+             'job_state': 'F'}
+        self.server.expect(JOB, a, extend='x', offset=5, id=jid)
+
+    def test_resource_invisible(self):
+        """
+        Test that value aggregation is same for invisible resources
+        """
+
+        # Set float and string_array to be invisible resource
+        attr = {}
+        attr['flag'] = 'ih'
+        self.server.manager(
+            MGR_CMD_SET, RSC, attr, id='foo_f', runas=ROOT_USER)
+        self.server.manager(
+            MGR_CMD_SET, RSC, attr, id='foo_str', runas=ROOT_USER)
+
+        hook_body = """
+import pbs
+e=pbs.event()
+j = e.job
+if j.in_ms_mom():
+    j.resources_used["foo_f"] = 2.114
+    j.resources_used["foo_str"] = '{"one":1,"two":2}'
+else:
+    j.resources_used["foo_f"] = 3.246
+    j.resources_used["foo_str"] = '{"two":2, "three":3}'
+"""
+
+        hook_name = "epi_invis"
+        a = {'event': "execjob_epilogue", 'enabled': 'True'}
+        self.server.create_import_hook(
+            hook_name,
+            a,
+            hook_body,
+            overwrite=True)
+
+        a = {'Resource_List.select': '3:ncpus=1',
+             'Resource_List.walltime': 10,
+             'Resource_List.place': 'scatter'}
+        j = Job(TEST_USER)
+        j.set_attributes(a)
+        j.set_sleep_time("5")
+        jid = self.server.submit(j)
+
+        # Verify that values are accumulated for float and string array
+        a = {'resources_used.foo_f': '8.606'}
+        self.server.expect(JOB, a, extend='x', offset=5, id=jid)
+
+        foo_str_dict_in = eval('{"one":1, "two":2, "three":3}')
```

Comment:
Same as a previous comment about setting the dictionary directly vs using eval()